### ISSUE TITLE
Create a way for auto-documenting rules in oxlint plugins

### ISSUE THREAD

**Issue by spaceemotion** [oxc-project/oxc#21318]
_Opened because of this X thread: https://x.com/TheAlexLichter/status/2042597682938556430_ 

We're having great success with custom rules so far, however, we have quite a bit of boilerplate to make them "nicer" to discover/use. Most of the popular ESLint packages have this nice overview in their README as well as a per-rule detailed rule information file with the good/bad examples.

Maybe oxlint could help out with: allowing rule meta data or a generally easier way to set up tests combined with auto-generating documentation? Right now, we generate our individual READMEs from the test cases with a bit of a janky script.

Having a native way like `import { RuleDocumentor } from "oxlint/plugin-dev"` or something, would be great!

1. Generates a nice documentation file with established conventions:
   - Sections for good vs bad
   - A table with supported options
   - Auto-Fix-level Indicators
2. Generates a nice "Table of rules" to put into a general README.

We currently also split our own rules into types (e.g. security, vitest, ...) so not sure if that should be part of this as well.

# Potential API
Just jotting some ideas down here:

```ts
interface RuleDocumentation {
  raw: string;
  name: string;
  description: string;
  examples: Array<{
    type: 'good' | 'bad';
    code: string;
  }>;
  options: unknown; // ???
}

interface RuleDocumentor {
  forRule(rule): RuleDocumentation
  forPlugin(plugin): {
    raw: string;
    rules: RuleDocumentation[];
  };
}
```
